src/main/java/com/marklogic/kafka/connect/sink/DefaultSinkRecordConverter.java

```diff
@@ -94,22 +100,29 @@ private DocumentMetadataHandle buildAdditionalMetadata(SinkRecord sinkRecord) {
         if (this.addTopicToCollections) {
             metadata.getCollections().add(sinkRecord.topic());
         }
-        if (this.includeKafkaMetadata) {
+        if (this.includeKafkaMetadata || this.includeKafkaHeaders) {
             DocumentMetadataHandle.DocumentMetadataValues values = metadata.getMetadataValues();
-            Object key = sinkRecord.key();
-            if (key != null) {
-                values.add("kafka-key", key.toString());
-            }
-            values.add("kafka-offset", sinkRecord.kafkaOffset() + "");
-            Integer partition = sinkRecord.kafkaPartition();
-            if (partition != null) {
-                values.add("kafka-partition", partition + "");
+            if (this.includeKafkaMetadata) {
```

Comment:
I think this is worth a private method, the current method was already a bit long - e.g. `addKafkaMetadataToDocumentMetadata`.